src/app/GitUI/UserControls/FileStatusList.cs

```diff
@@ -313,39 +339,31 @@ public override ContextMenuStrip ContextMenuStrip
         public Func<ObjectId?, string>? DescribeRevision { get; set; }
 
         public bool FilterFocused => FilterComboBox.Focused;
-        public bool SearchFocused => SearchComboBox.Focused;
+        public bool SearchFocused => SearchCommitGitGrepComboBox.Focused;
 
         /// <summary>
-        /// Enable Search Commit combobox for the file list.
-        /// Refresh of list must be called explicitly.
+        ///  Indicates whether the git-grep search functionality is enabled for this control.
         /// </summary>
-        public bool SearchEnabledForList
-        {
-            private get => _searchEnabledForList;
-            set
-            {
-                _searchEnabledForList = value;
-                EnableSearchForList(value && AppSettings.ShowSearchCommit.Value);
-            }
-        }
+        [DefaultValue(false)]
+        public bool CanUseSearchCommitGitGrep { get; set; }
 
-        private void EnableSearchForList(bool enable)
+        public void ToggleSearchCommitGitGrep(bool visible)
```

Comment:
Also align name with `EnableSearchBoxFunc` in FormSearchCommitGitGrep, maybe `SetSearchCommitGitGrepFunc`
(the original name was choosen in analogy with the Find popup)

```suggestion
        public void SetSearchCommitGitGrep(bool visible)
```